api/client/src/main/java/jakarta/websocket/Session.java

```diff
@@ -173,21 +173,63 @@
 
     /**
      * Return the number of milliseconds before this conversation may be closed by the container if it is inactive, i.e.
-     * no messages are either sent or received in that time.
+     * no messages are either sent or received in that time. A value that is zero or negative indicates that this
+     * timeout will not be used. The read specific ({@link #getMaxReadIdleTimeout()} and/or write specific {@link
+     * #getMaxWriteIdleTimeout()} timeouts may still apply.
      *
      * @return the timeout in milliseconds.
      */
     long getMaxIdleTimeout();
 
     /**
      * Set the non-zero number of milliseconds before this session will be closed by the container if it is inactive, ie
-     * no messages are either sent or received. A value that is 0 or negative indicates the session will never timeout
-     * due to inactivity.
+     * no messages are either sent or received. A value that is 0 or negative indicates that this timeout will not be
+     * used. The read specific ({@link #getMaxReadIdleTimeout()} and/or write specific {@link #getMaxWriteIdleTimeout()}
+     * timeouts may still apply.
      *
      * @param milliseconds the number of milliseconds.
      */
     void setMaxIdleTimeout(long milliseconds);
 
+    /**
+     * Return the number of milliseconds before this conversation may be closed by the container if no messages are
+     * received in that time. A value that is zero or negative indicates that this timeout will not be used. The general
```

Comment:
Complete message? (all data frames?)
Partial message? (some data frames? an incomplete data frame?)
Control frame?
Reset on receive of traffic?
Or reset on delivery of data frame? or any frame? or even auto-fragmented frame?
What if the connection has higher level activity not within websocket (eg: http/2 stream specific details, or http/3 stream specific details)?